src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/managehearings/HearingsAboutToSubmitHandlerTest.java

```diff
@@ -49,57 +54,71 @@ void testCanHandle() {
                 .build())
             .build();
 
         FinremCallbackRequest request = FinremCallbackRequestFactory.from(Long.parseLong(caseReference),
             CaseType.CONTESTED, caseData);
 
+        doAnswer(invocation -> {
+
+            UUID workingHearingID = UUID.randomUUID();
+            ManageHearingsCollectionItem manageHearingsCollectionItem = ManageHearingsCollectionItem.builder()
+                .id(workingHearingID)
+                .value(hearingToAdd)
+                .build();
+
+            ManageHearingDocumentsCollectionItem manageHearingDocumentsCollectionItem = ManageHearingDocumentsCollectionItem
+                .builder()
+                .id(UUID.randomUUID())
+                .value(ManageHearingDocument.builder()
+                    .hearingId(workingHearingID)
+                    .hearingDocument(CaseDocument.builder()
+                        .categoryId("categoryId")
+                        .documentUrl("documentUrl")
+                        .documentFilename("HearingNotice.pdf")
+                        .documentBinaryUrl("documentBinaryUrl")
+                        .uploadTimestamp(LocalDateTime.now())
+                        .build())
+                    .build())
+                .build();
+
+            FinremCaseDetails details = invocation.getArgument(0);
+            ManageHearingsWrapper wrapper = details.getData().getManageHearingsWrapper();
+
+            wrapper.setHearings(List.of(manageHearingsCollectionItem));
+            wrapper.setHearingDocumentsCollection(List.of(manageHearingDocumentsCollectionItem));
+            wrapper.setWorkingHearing(null);
+            wrapper.setWorkingHearingId(workingHearingID);
+            return null;
+        }).when(manageHearingActionService)
+            .performAddHearing(request.getCaseDetails(), AUTH_TOKEN);
+
+        // Act
         var response = manageHearingsAboutToSubmitHandler.handle(request, AUTH_TOKEN);
         var responseManageHearingsWrapper = response.getData().getManageHearingsWrapper();
+        var hearingDocumentAdded = responseManageHearingsWrapper.getHearingDocumentsCollection().getFirst();
+        var hearingId = responseManageHearingsWrapper.getHearings().getFirst().getId();
 
+        //Assert
         assertThat(responseManageHearingsWrapper.getHearings())
             .extracting(ManageHearingsCollectionItem::getValue)
             .contains(hearingToAdd);
+        assertThat(hearingDocumentAdded.getValue().getHearingId()).isEqualTo(hearingId);
+        assertThat(hearingDocumentAdded.getValue().getHearingDocument().getDocumentFilename())
+            .isEqualTo("HearingNotice.pdf");
+        assertThat(responseManageHearingsWrapper.getWorkingHearingId()).isEqualTo(hearingId);
         assertThat(responseManageHearingsWrapper.getWorkingHearing()).isNull();
-        assertThat(responseManageHearingsWrapper.getManageHearingsActionSelection()).isNull();
     }
 
     private Hearing createHearingToAdd() {
```

Comment:
Not to worry for this PR, but this is a big test to follow.

If we're in this test again for any reason, will look to refactor doAnswer to a method called something like mockPerformAddHearingResponse, then instantiate that in a setup.